src/components/Recoil_RecoilRoot.react.js

```diff
@@ -120,7 +120,7 @@ function Batcher(props: {setNotifyBatcherOfChange: (() => void) => void}) {
 }
 
 if (__DEV__) {
-  if (!window.$recoilDebugStates) {
+  if (typeof window !== 'undefined' && !window.$recoilDebugStates) {
```

Comment:
Should we make this a single if?

```
if (__DEV__ && typeof...)
```

Reply:
@mondaychen - for the sync scripts does the `__DEV__` check need to be by itself?

Reply:
No. `__DEV__` can be used like a simple boolean variable.

Reply:
FYI - We have an internal lint warning to keep `__DEV__` checks on their own line for clarity since they are crushed in production.

Reply:
Cool. What I said only applies to our github build process.